### PR TITLE
[no ticket][risk=no] Puppeteer new cohort-edit.spec

### PR DESCRIPTION
**Pull request by aweng98**
New `cohort-edit.spec` test that covers testing of cohort actions in Cohort Build page:
- Discard changes
- Rename group name
- Insert new group
- Review Sets for cohort
- Delete cohort
- Save as cohort

Screen Shot 2021-05-06 at 2 31 06 PM

Goal: Increase test coverage; maximize single workspace usage (one workspace for all tests); group the testing of Cohort Build page in one file.

Deleted `cohort-duplicate.spec` because it is obsolete (duplicate testing).

Created many new functions to support new tests, improved code and pages synchronization. 

Future Plan:
- Delete `cohorts-rename.spec` (testing already covered in `cohort-edit.spec`)
- Expand `cohorts-create.spec` to include testing of all Criteria.

### CODE REVIEW

File: e2e/utils/str-utils.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function makeWorkspaceName(): string {
 
 export function makeRandomName(prefix?: string): string {
   prefix = prefix || 'aoutest';
-  return `${prefix}-${Math.floor(Math.random() * 10000000)}`;
+  return `${prefix}-${Math.floor(Math.random() * 10000000000)}`;
```
Comment (aweng98) [inline e2e/utils/str-utils.ts:37]: Make random number bigger to prevent name conflict issue.

File: e2e/tests/cohorts/cohorts-create.spec.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-import Link from 'app/element/link';
 import CohortBuildPage, { FieldSelector } from 'app/page/cohort-build-page';
 import WorkspaceDataPage from 'app/page/workspace-data-page';
 import { findOrCreateWorkspace, signInWithAccessToken } from 'utils/test-utils';
 import { waitForText } from 'utils/waits-utils';
+import { numericalStringToNumber } from 'utils/str-utils';
+import CohortActionsPage from 'app/page/cohort-actions-page';
 
-describe('Cohorts', () => {
+describe('Create Cohort', () => {
```
Comment (aweng98) [inline e2e/tests/cohorts/cohorts-create.spec.ts:9]: `cohorts-create.spec` will be reworked to include more tests.

**Review by dolbeew — APPROVED** [review]
New tests look good, thanks.

**Comment by aweng98** [comment]
Thank you.

**Pull request merged**